R/btw_client.R

```diff
@@ -135,8 +135,7 @@ btw_client <- function(..., client = NULL, tools = NULL, path_btw = NULL) {
   client$set_system_prompt(paste(sys_prompt, collapse = "\n"))
 
   if (!skip_tools) {
-    maybe_quiet <- if (is.null(tools)) suppressMessages else I
-    maybe_quiet(btw_register_tools(client, tools = config$tools))
+    client$set_tools(tools = btw_tools(config$tools))
```

Comment:
Totally agree with your top-level assessment that we should use and document the use of the ellmer API, i.e. using `$set_tools()` and not our own `btw_register_tools()`.

That said, I _do think_ we should wrap that action in `btw_client()`

```suggestion
    client$set_tools(tools = c(client$get_tools(), btw_tools(config$tools)))
```